### DISCUSSION TITLE
Making translations actually work

### DISCUSSION THREAD

**Discussion by Miniontoby** [pelican-dev/panel#79] [Ideas]
Hi,

I had been working on making the translations actually be usable for pterodactyl lately, but since this is a fork that is more active, I was thinking that it might be more useful here.

I would be glad to take this task on me, if wanted.

Kind regards,
Miniontoby

Update:
It seems like the `feature/filament` branch includes the new panel. 
I guess this is going to be the new frontend, so I guess I will have to make that translatable instead

Update:
Current patch file can be found at https://github.com/pelican-dev/panel/pull/140.diff

**Comment by Boy132** [comment]
Correct. `feature/filament` is the development branch for the new backend ui made with [filament](https://filamentphp.com/).
The frontend ui will be converted eventually too.
Localization is planned.

**Reply by Miniontoby** [reply]
I am already working on making the new filament page translatable, now.

I might even make a PR when I am done...

**Comment by Poseidon281** [comment]
There is a crowdin page. So if you have source files that need to be translated they can be added to the crowdin page for translation

**Reply by Miniontoby** [reply]
Yeah, I saw the crowdin page, also joined it now.

But would it be fine if I just commit my code and then it gets uploaded using the crowdin cli?

I got English and Dutch at the same time for the `feature/filament` branch.

Translated the Dashboard and most of the other pages/resources also.

Also included changes to the default ptero frontend and backend, but focussed mostly on the new filament views.


*Also, most of the already existing keys are not used in the panel itself, so most of them are not useful to translate...*

**Comment by Poseidon281** [comment]
> Yeah, I saw the crowdin page, also joined it now.
> 
> But would it be fine if I just commit my code and then it gets uploaded using the crowdin cli?
> 
> I got English and Dutch at the same time for the `feature/filament` branch.
> 
> Translated the Dashboard and most of the other pages/resources also.
> 
> Also included changes to the default ptero frontend and backend, but focussed mostly on the new filament views.
> 
> _Also, most of the already existing keys are not used in the panel itself, so most of them are not useful to translate..._

What have you translated right now/created a translate for? Just so i dont do the work double. Maybe create a draft PR just so people can see your progress and to avoid double work

**Reply by Miniontoby** [reply]
Every single sentence that I have added into the panel is translated into Dutch (except for the whole `strings.php`)

The problem that I have with draft PR's is that I cannot keep my stuff within 1 commit, which I personally prefer. Mostly because I can then easily make a patch file so I can share it with others. 

How ever I can put the patch file in here and update it frequently... if that is fine/wanted?

**Reply by Poseidon281** [reply]
Have you done /resources/views already? If not i am going to work on that?

**Reply by Miniontoby** [reply]
As far as I know `/resources/views` is only the `/resources/views/filament` really translatable. And that is only the main page and I already made that one translatable. 

However I do not know what you mean with "work on that", does that mean you translate it or you make it translatable?

**Reply by Poseidon281** [reply]
#118 I am currently working on this. Making it translatable and then when it is added to crowdin it can be translated and i can translate the Dutch part and some german

**Reply by Miniontoby** [reply]
Oh you have translated the old admin layout... seems useless, since filament will be the replacement for /admin...

But sure. 

And I guess I should make my branch, tho

**Reply by Poseidon281** [reply]
Oh yes you are right. I forgot about that. Well oke. And you made filament translatable already right? Thanks for the reminder.

**Reply by Miniontoby** [reply]
At least the main page of filament, yes

**Reply by Poseidon281** [reply]
> Every single sentence that I have added into the panel is translated into Dutch (except for the whole `strings.php`)
> 
> The problem that I have with draft PR's is that I cannot keep my stuff within 1 commit, which I personally prefer. Mostly because I can then easily make a patch file so I can share it with others.
> 
> How ever I can put the patch file in here and update it frequently... if that is fine/wanted?

If you could do that that would be really nice. Mostly to avoid double work and things like that

**Reply by Miniontoby** [reply]
I will be making my PR really soon.

But the patch file for now is located at https://github.com/pelican-dev/panel/discussions/79#discussioncomment-9175113

It is nearly complete, except for the fields that have names that aren't yet an key in the translation files, just need to take some time to manually add all of them...

**Reply by Miniontoby** [reply]
Oke, it is a bit too much to keep up, so I guess I will make the patch work again and then just make my PR and fix the rest in a later commit.

Just because this repo is too active... ;)

**Reply by Poseidon281** [reply]
Yes just your work until now make a PR and I can help with making the translation file to. I can finish making it. Have a bit of spare time.

**Reply by Miniontoby** [reply]
Oke, deleted the patch file since I made #140

**Reply by Miniontoby** [reply]
@Poseidon281 
> Yes just your work until now make a PR and I can help with making the translation file to. I can finish making it. Have a bit of spare time.

I also have enough spare time, tho. And I like to be part of this as a developer.
But yeah, we could of course do this together... Possibly the best to make like a todo board so we can decide who does what, then.
You could join my github team, since you are Dutch, so if you would like to and then we can do a collab PR, for last bits and pieces?

**Reply by Poseidon281** [reply]
I can create a TODO board as a Trello and for now i can join your Github Team.

**Reply by Miniontoby** [reply]
Yeah, trello could also, but why not just use Github, since it already has it in there

**Comment by Miniontoby** [comment]
Oke, deleted the patch file since I made #140

New patch is https://github.com/pelican-dev/panel/pull/140.diff

**Comment by lancepioch** [comment]
Thanks, we need to revisit translations from the ground up!
